src/op_handlers/far_call.rs

```diff
@@ -204,7 +204,9 @@ pub fn far_call(
 
     let exception_handler = opcode.imm0 as u64;
 
-    let abi = get_far_call_arguments(src0.value);
+    let mut abi = get_far_call_arguments(src0.value);
+    abi.is_constructor_call = abi.is_constructor_call && vm.current_context()?.is_kernel();
+    abi.is_system_call = abi.is_system_call && is_kernel(&contract_address);
```

Comment:
Is there any reason why lines `208` and `209` are not being handled inside of `get_far_call_arguments()`?

Reply:
Not really, just that the vm2 does it this way, however that would involve passing the vm and the contract address as parameters of `get_far_call_arguments`, i think is simpler this way